packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/tool-list/PaletteQuickAccessToolBar.tsx

```diff
@@ -0,0 +1,150 @@
+import Tooltip from '@mui/material/Tooltip';
+import { makeStyles } from 'tss-react/mui';
+import { PinIcon } from '../../../icons/PinIcon';
+import { UnpinIcon } from '../../../icons/UnpinIcon';
+import { Tool } from '../../Tool';
+import { useAdjustSize } from '../../adjust-size/useAdjustSize';
+import { useFadeDiagramElements } from '../../fade/useFadeDiagramElements';
+import { usePinDiagramElements } from '../../pin/usePinDiagramElements';
+import { isSingleClickOnDiagramElementTool } from '../Palette';
+import { PaletteQuickAccessToolBarProps } from './PaletteQuickAccessToolBar.types';
+
+const useStyle = makeStyles()((theme) => ({
+  quickAccessTools: {
+    display: 'flex',
+    flexWrap: 'nowrap',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+    overflowX: 'auto',
+  },
+  toolIcon: {
+    width: theme.spacing(4.5),
+    color: theme.palette.text.primary,
+  },
+}));
+
+export const PaletteQuickAccessToolBar = ({
+  node,
+  diagramElementId,
+  palette,
+  onToolClick,
+  paletteToolComponents,
+  hideableDiagramElement,
+  x,
+  y,
+}: PaletteQuickAccessToolBarProps) => {
+  const { classes } = useStyle();
+  const { fadeDiagramElements } = useFadeDiagramElements();
+  const { pinDiagramElements } = usePinDiagramElements();
+  const { adjustSize } = useAdjustSize();
+  let pinUnpinTool: JSX.Element | undefined;
+  let adjustSizeTool: JSX.Element | undefined;
+  if (node) {
+    pinUnpinTool = node.data.pinned ? (
+      <Tooltip title="Unpin element" key={'tooltip_unpin_element_tool'}>
+        <IconButton
+          className={classes.toolIcon}
+          size="small"
+          aria-label="Unpin element"
+          onClick={() => pinDiagramElements([diagramElementId], !node.data.pinned)}
+          data-testid="Unpin-element">
+          <UnpinIcon fontSize="small" />
+        </IconButton>
+      </Tooltip>
+    ) : (
+      <Tooltip title="Pin element" key={'tooltip_pin_element_tool'}>
+        <IconButton
+          className={classes.toolIcon}
+          size="small"
+          aria-label="Pin element"
+          onClick={() => pinDiagramElements([diagramElementId], true)}
+          data-testid="Pin-element">
+          <PinIcon fontSize="small" />
+        </IconButton>
+      </Tooltip>
+    );
+    adjustSizeTool = (
+      <Tooltip title="Adjust size" key={'tooltip_adjust_element_tool'}>
+        <IconButton
+          className={classes.toolIcon}
+          size="small"
+          aria-label="Adjust element"
+          onClick={() => adjustSize(diagramElementId)}
+          data-testid="adjust-element">
+          <AdjustIcon fontSize="small" />
+        </IconButton>
+      </Tooltip>
+    );
+  }
+
+  const invokeFadeDiagramElementTool = () => {
+    fadeDiagramElements([diagramElementId], true);
+  };
+  const quickAccessToolComponents: JSX.Element[] = [];
+  palette?.quickAccessTools
+    .filter(isSingleClickOnDiagramElementTool)
```

Comment:
Why accept any kind of tool as quick access tools in the GraphQL API if only "single click on diagram element tools" are supported?

Reply:
Indeed we should not restrict to isSingleClickOnDiagramElementTool. The palette is also intended to be displayed on edge tools